src/index.ts

```diff
@@ -1,48 +1,65 @@
 import * as qs from 'qs';
+import * as packageJSON from '../package.json';
+
 import { DOMAIN, MY_ACCOUNT, VAULT } from './endpoints';
 
 interface Config {
   clientId: string;
-  isTestEnvironment?: boolean;
   scope: string[];
+  isTestEnvironment?: boolean;
   redirectUri?: string;
   continueTo?: string;
-  responseType?: string;
+  responseType?: 'code' | 'token';
   locale?: string;
   state?: string;
 }
 
 interface Params {
   client_id: string;
   redirect_uri: string;
-  continue: string;
   response_type: string;
   scope: string;
-  back?: string;
+  continue?: string;
   locale?: string;
   state?: string;
 }
 
+interface Domains {
+  vault: string;
+  myaccount: string;
+}
+
 interface VaultOptions {
   backTo?: string;
   newTab?: boolean;
 }
 
-interface MyaccountOptions {
+interface MyAccountOptions {
   backTo?: string;
   newTab?: boolean;
   email?: string;
   authPage?: string;
+  showAuthToggle?: boolean;
 }
 
-const encodeConfig = (configs: object): string => {
-  // My Account separate key/values using semicolon
-  return encodeURIComponent(qs.stringify(configs, { delimiter: ';' }));
+function removeOAuth2Params(params: Params) {
```

Comment:
What's that?
If we need a different set of params I'm sure there is a way better solution

Reply:
I assumed initially, you pass the OAuth related config via `init` method is so that there is no need to feed in the same parameters for calling `authorize` multiple times.

However, those params are not needed for other method call.

Reply:
This is bad architecture. Need to be fixed asap

Reply:
how would you like it to be?
add oauth via authorize instead of init?
that will requires deprecation notes and same method will still stays there to remove those.

Or no backward compatibility?

Reply:
The opposite. Have two sets of params, which can be merged where needed

Reply:
not sure if I understand, but what is the point of sending things like scope, response type and redirect uri to settings or vault?

Reply:
If you have two sets of params you won't be sending those.
So instead of removing the ones you don't want. Merge the two sets.